### PR TITLE
chore(infra): add rust codecov

### LINKED ISSUE #7090: integrate with code coverage
we did lots of refactor and may have some legacy code which is not removed and we also could have some code which should be covered by tests, use code coverage service could help us spot these issues

### PR DESCRIPTION
**Pull request by SoonIter**
## Summary

close #7090 

We may also need the coverage of js jest test.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -320,6 +320,64 @@ jobs:
...
+          shared-key: check
+
+      - name: Install llvm-tools-preview
+        run: rustup component add llvm-tools-preview
+      - uses: cargo-bins/cargo-binstall@v1.10.2
+      - run: cargo binstall --no-confirm cargo-llvm-cov
+
+      - name: Run test coverage
+        # reason for excluding https://github.com/napi-rs/napi-rs/issues/2200
+        run: cargo llvm-cov --workspace --exclude rspack_binding_options --exclude rspack_node --lcov --output-path lcov.info
```
Comment (chenjiahan) [inline .github/workflows/ci.yml:347]: Why do we need to run test in two different jobs (rust_test and rust_test_codecov), can we unify them?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -320,6 +320,64 @@ jobs:
...
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: codecov
+          path: lcov.info
+
+  rust_test_codecov_upload:
```
Comment (chenjiahan) [inline .github/workflows/ci.yml:355]: Why split rust_test_codecov and rust_test_codecov_upload, can we merge them?

**Comment by SoonIter** [comment]
😂 7.05%

image

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SoonIter) [reply .github/workflows/ci.yml:327]: done, merged in `rust_test` with `cargo test`

File: .github/workflows/ci.yml (hunk 2 above)
Comment (SoonIter) [reply .github/workflows/ci.yml:336]: `# codecov often fails, use another workflow for retry` 
https://github.com/web-infra-dev/rspack-resolver/blob/16eb56eef68d651edec703b94022ff7f3e46d585/.github/workflows/codecov.yml#L44

Migrated from `rspack-resolver` The generated file is huge, and the api of codecov does not seem to be very stable.

**Comment by SoonIter** [comment]
can be added with rust/js unit test refactor
There is no need of codecov at present

We are looking for a way to calculate the coverage rate of the **intergration/e2e test**

**Pull request closed without merging**